Bio/PDB/Entity.py

```diff
@@ -13,20 +13,37 @@
 
 from collections import deque
 from copy import copy
+from typing import Any, Dict, Generic, List, Optional, Protocol, TypeVar
```

Comment:
I'm not familiar with Protocol and TypeVar here, but these Entity changes aside it all looks fine to me.

Reply:
I found a slightly more elegant solution which doesn't need `Protocol`. Can bind directly to `Union["Entity", "Atom"]` and to `Optional["Entity"]` instead.